atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/SweepBatchAccumulatorTest.java

```diff
@@ -0,0 +1,238 @@
+            assertThat(batch.writes()).isEmpty();
+            assertThat(batch.dedicatedRows()).isEqualTo(NO_DEDICATED_ROWS);
+            assertThat(batch.lastSweptTimestamp()).isEqualTo(SWEEP_TIMESTAMP - 1);
+        });
+        assertThat(batchWithPartitionInfo.finePartitions()).isEmpty();
+    }
+
+    @Test
+    public void oneBatchAccumulatesToItselfAndLogsPartition() {
+        SweepBatch sweepBatch = SweepBatch.of(
+                ImmutableList.of(WRITE_INFO_1),
+                DEDICATED_ROWS_1,
+                PROGRESS_TIMESTAMP + 200);
+        accumulator.accumulateBatch(sweepBatch);
+
+        SweepBatchWithPartitionInfo batchWithPartitionInfo = accumulator.toSweepBatch();
+        assertThat(batchWithPartitionInfo.sweepBatch()).isEqualTo(sweepBatch);
+        assertThat(batchWithPartitionInfo.finePartitions())
+                .containsExactly(SweepQueueUtils.tsPartitionFine(PROGRESS_TIMESTAMP + 200));
+    }
+
+    @Test
+    public void mergesTwoBatchesWithDistinctWriteInfo() {
+        accumulator.accumulateBatch(SweepBatch.of(
+                ImmutableList.of(WRITE_INFO_1),
+                NO_DEDICATED_ROWS,
+                PROGRESS_TIMESTAMP + 177));
+        accumulator.accumulateBatch(SweepBatch.of(
+                ImmutableList.of(WRITE_INFO_2),
+                NO_DEDICATED_ROWS,
+                PROGRESS_TIMESTAMP + 288));
+
+        SweepBatchWithPartitionInfo batchWithPartitionInfo = accumulator.toSweepBatch();
+        assertThat(batchWithPartitionInfo.sweepBatch()).satisfies(batch -> {
+            assertThat(batch.writes()).containsExactlyInAnyOrder(WRITE_INFO_1, WRITE_INFO_2);
+            assertThat(batch.dedicatedRows()).isEqualTo(NO_DEDICATED_ROWS);
+            assertThat(batch.lastSweptTimestamp()).isEqualTo(PROGRESS_TIMESTAMP + 288);
+        });
+        assertThat(batchWithPartitionInfo.finePartitions()).isEqualTo(ImmutableSet.of(
+                SweepQueueUtils.tsPartitionFine(PROGRESS_TIMESTAMP + 177),
+                SweepQueueUtils.tsPartitionFine(PROGRESS_TIMESTAMP + 288)));
+    }
+
+    @Test
+    public void mergesDedicatedRows() {
+        accumulator.accumulateBatch(SweepBatch.of(
+                ImmutableList.of(),
+                DEDICATED_ROWS_1,
+                PROGRESS_TIMESTAMP + 177));
+        accumulator.accumulateBatch(SweepBatch.of(
+                ImmutableList.of(),
+                DEDICATED_ROWS_2,
+                PROGRESS_TIMESTAMP + 288));
+
+        SweepBatchWithPartitionInfo batchWithPartitionInfo = accumulator.toSweepBatch();
+        assertThat(batchWithPartitionInfo.sweepBatch()).satisfies(batch -> {
+            assertThat(batch.dedicatedRows()).isEqualTo(DedicatedRows.of(
+                    ImmutableList.of(SWEEPABLE_CELLS_ROW_1, SWEEPABLE_CELLS_ROW_2)));
+            assertThat(batch.lastSweptTimestamp()).isEqualTo(PROGRESS_TIMESTAMP + 288);
+        });
+        assertThat(batchWithPartitionInfo.finePartitions()).isEqualTo(ImmutableSet.of());
```

Comment:
This is actually a bug, we should still delete the rows with the references to the dedicated rows!

Reply:
Yep, makes sense. I've modified the accumulator to map the rows to their partitions too.